src/raftstore/store/worker/region.rs

```diff
@@ -331,6 +336,29 @@ impl SnapContext {
         Ok(())
     }
 
+    // check the number of files at level 0 to avoid write stall after ingesting sst,
+    // return indicate whether ingest will cause write stall or not.
+    fn ingest_maybe_stall(&self) -> bool {
+        for cf in SNAPSHOT_CFS {
+            // no need to check lock cf
+            if plain_file_used(cf) {
+                continue;
+            }
+
+            let handle = rocksdb::get_cf_handle(&self.kv_db, cf).unwrap();
+            if let Some(n) = get_cf_num_files_at_level(&self.kv_db, handle, 0) {
+                let options = self.kv_db.get_options_cf(handle);
+                let slowdown_triggers = options.get_level_zero_slowdown_writes_trigger();
+                // Leave enough buffer to tolerate heavy write workload,
+                // which may flush some memtables in a short time.
+                if n > slowdown_triggers as u64 / 2
```

Comment:
@Connor1996 I change this logic a little bit, because according to my tests, the original implementation can still get stall easily during heavy write workload.

Reply:
would this make apply snapshot process starved in heavy write workload?

Reply:
I think it will not be a problem if the `slowdown_trigger` is not too small. It's reasonable to slow down the snapshot ingestion during heavy write workload.